tests/validate/bandwidth_test/src/host.cpp

```diff
@@ -151,102 +161,175 @@ int main(int argc, char** argv) {
 
     double max_throughput = 0;
     int reps = stoi(iter_cnt);
+    if (num_kernel_ddr) {
+        //Starting at 4K and going up to 16M with increments of power of 2
+        for (uint32_t i = 4 * 1024; i <= 16 * 1024 * 1024; i *= 2) {
+            unsigned int data_size = i;
+
+            if (xcl::is_emulation()) {
+                reps = 2; // reducing the repeat count to 2 for emulation flow
+                // Running only upto 8K for emulation flow
+                if (data_size > 8 * 1024) {
+                    break;
+                }
+            }
 
-    for (uint32_t i = 4 * 1024; i <= 16 * 1024 * 1024; i *= 2) {
-        unsigned int DATA_SIZE = i;
+            unsigned int vector_size_bytes = data_size;
+            std::vector<unsigned char, aligned_allocator<unsigned char> > input_host(data_size);
+            std::vector<unsigned char, aligned_allocator<unsigned char> > output_host[num_kernel_ddr];
 
-        if (xcl::is_emulation()) {
-            reps = 2;
-            if (DATA_SIZE > 8 * 1024) break;
-        }
+            for (int i = 0; i < num_kernel_ddr; i++) {
+                output_host[i].resize(data_size);
+            }
+            for (uint32_t j = 0; j < data_size; j++) {
+                input_host[j] = j % 256;
+            }
 
-        unsigned int vector_size_bytes = DATA_SIZE;
-        std::vector<unsigned char, aligned_allocator<unsigned char> > input_host(DATA_SIZE);
-        std::vector<unsigned char, aligned_allocator<unsigned char> > output_host[NUM_KERNEL];
+            // Initializing output vectors to zero
+            for (int i = 0; i < num_kernel_ddr; i++) {
+                std::fill(output_host[i].begin(), output_host[i].end(), 0);
+            }
 
-        for (int i = 0; i < NUM_KERNEL; i++) {
-            output_host[i].resize(DATA_SIZE);
-        }
-        for (uint32_t j = 0; j < DATA_SIZE; j++) {
-            input_host[j] = j % 256;
-        }
+            std::vector<cl::Buffer> input_buffer(num_kernel_ddr);
+            std::vector<cl::Buffer> output_buffer(num_kernel_ddr);
+
+            // These commands will allocate memory on the FPGA. The cl::Buffer objects
+            // can
+            // be used to reference the memory locations on the device.
+            // Creating Buffers
+            for (int i = 0; i < num_kernel_ddr; i++) {
+                OCL_CHECK(err,
```

Comment:
Hmm...format is inconsistent with the rest of the method.  Might wish to put it back on one line for this isn't kernel C code.

Reply:
We have a format.sh script that is used for code formatting. The host code pushed in PR is after running the format script.
/wrk/xhdhdnobkup1/hatchuta/public/format.sh